### PR TITLE
Adding http client docs

### PR DESCRIPTION
**Pull request by srtfisher**

### CODE REVIEW

File: features/http-client.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# HTTP Client
+
+[[toc]]
+
+## Introduction
+
+Mantle provides a fluent and expressive wrapped around the [WordPress HTTP
```
Comment (mogmarsh) [inline features/http-client.md:7]: this line doesn't quite read right to me

File: features/http-client.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+::: details Future Plans
+In the future, the HTTP Client will be abstracted to support the Guzzle HTTP Client
+and multipart requests. This also allows parallel requests to be made.
+:::
+
+## Making Requests
+
+Making requests with the HTTP Client can be done using any HTTP verb such as
+`get`, `head`, `post`, `put`, `delete` and `patch`. Requests can be made using
+the `Http` or instantiating `Mantle\Http\Client\Http_Client` directly:
```
Comment (mogmarsh) [inline features/http-client.md:22]:
```suggestion
the `Http` class or instantiating `Mantle\Http\Client\Http_Client` directly:
```

**Review by mogmarsh — APPROVED** [review]
a couple small things, otherwise 👍

**Comment by srtfisher** [comment]
Manually pushed these up

**Pull request merged**